core/src/services/d1/backend.rs

```diff
@@ -0,0 +1,376 @@
+
+        let root = normalize_root(
+            self.root
+                .clone()
+                .unwrap_or_else(|| "/".to_string())
+                .as_str(),
+        );
+        Ok(D1Backend::new(Adapter {
+            authorization,
+            account_identifier,
+            database_identifier,
+            endpoint,
+            client,
+            table,
+            key_field,
+            value_field,
+        })
+        .with_root(&root))
+    }
+}
+
+pub type D1Backend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    authorization: Option<String>,
+    account_identifier: String,
+    database_identifier: String,
+
+    endpoint: String,
+    client: HttpClient,
+
+    table: String,
+    key_field: String,
+    value_field: String,
+}
+
+impl Debug for Adapter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("D1Adapter");
+        ds.field("endpoint", &self.endpoint);
+        ds.field("table", &self.table);
+        ds.field("key_field", &self.key_field);
+        ds.field("value_field", &self.value_field);
+        ds.finish()
+    }
+}
+
+impl Adapter {
+    fn create_d1_query_request(&self, sql: &str, params: Vec<Value>) -> Result<Request<AsyncBody>> {
+        let p = format!(
+            "/accounts/{}/d1/database/{}/query",
+            self.account_identifier, self.database_identifier
+        );
+        let url: String = format!("{}{}", self.endpoint, percent_encode_path(&p));
+
+        let mut req = Request::post(&url);
+        if let Some(auth) = &self.authorization {
+            req = req.header(header::AUTHORIZATION, auth);
+        }
+        req = req.header(header::CONTENT_TYPE, "application/json");
+
+        let json = serde_json::json!({
+            "sql": sql,
+            "params": params,
+        });
+
+        let body_string = serde_json::to_string(&json).map_err(new_json_serialize_error)?;
+        let body_bytes = body_string.as_bytes().to_owned();
+        req.body(AsyncBody::Bytes(body_bytes.into()))
+            .map_err(new_request_build_error)
+    }
+}
+
+#[async_trait]
+impl kv::Adapter for Adapter {
+    fn metadata(&self) -> kv::Metadata {
+        kv::Metadata::new(
+            Scheme::D1,
+            &self.table,
+            Capability {
+                read: true,
+                write: true,
+                ..Default::default()
+            },
+        )
+    }
+
+    async fn get(&self, path: &str) -> Result<Option<Vec<u8>>> {
+        let query = format!(
+            "SELECT {} FROM {} WHERE {} = ? LIMIT 1",
+            self.value_field, self.table, self.key_field
+        );
+        let req = self.create_d1_query_request(&query, vec![path.into()])?;
+
+        let resp = self.client.send(req).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK | StatusCode::PARTIAL_CONTENT => {
+                let body = resp.into_body().bytes().await?;
```

Comment:
Please parse the response in a strong and explicit manner. The response for `SELECT {} FROM {} WHERE {} = ? LIMIT 1` should be clear and ensure accuracy.

Return unexpected error if the response is not as expected.

Reply:
@Xuanwo I've made updates to this code based on my understanding, but I still have some questions. Could you please provide me with some guidance?

Reply:
> And `in a strong and explicit manner` means?

Implement a `D1QueryResponse` and dervice `Deserialize` for with instead of playing with `serde_json::Map` and `serde_json::Value`.

Reply:
> Implement a `D1QueryResponse` and dervice `Deserialize` for with instead of playing with `serde_json::Map` and `serde_json::Value`.
I'll mark this as resolved, can you help review the latest code?